File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/provider/ParentsOperation.java
package app.zxtune.fs.provider;

import android.database.Cursor;
import android.net.Uri;

import androidx.annotation.Nullable;

import app.zxtune.fs.VfsObject;

class ParentsOperation implements AsyncQueryOperation {

  private final Uri uri;
  private final Resolver resolver;
  private final SchemaSource schema;

  ParentsOperation(Uri uri, Resolver resolver, SchemaSource schema) {
    this.uri = uri;
    this.resolver = resolver;
    this.schema = schema;
  }

  @Override
  public Cursor call() throws Exception {
    final VfsObject obj = resolver.resolve(uri);
    return ParentsCursorBuilder.makeParents(obj, schema);
  }

  @Nullable
  @Override
  public Cursor status() {
    return null;
  }
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/provider/ParentsOperation.kt
package app.zxtune.fs.provider

import android.database.Cursor
import android.net.Uri

internal class ParentsOperation(
    private val uri: Uri,
    private val resolver: Resolver,
    private val schema: SchemaSource
) : AsyncQueryOperation {
    override fun call() = ParentsCursorBuilder.makeParents(resolver.resolve(uri), schema)

    override fun status(): Cursor? = null
}
